### PR TITLE
[IOS-616]: Screen protection flow improvements

### PR DESCRIPTION
**Pull request by yuliaveres**
https://toshiapp.atlassian.net/browse/IOS-616

### CODE REVIEW

File: Toshi/Supporting Files/AppDelegate.swift [hunk 1]
```diff
@@ -263,14 +264,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     // Screen protection
 
-    private func activateScreenProtection() {
-
-        guard screenProtectionWindow == nil else { return }
-
+    private lazy var screenProtectionWindow: UIWindow = {
         let window = UIWindow()
         window.isHidden = true
         window.backgroundColor = .clear
```
Comment (Elland) [inline Toshi/Supporting Files/AppDelegate.swift]: if it's transparent, how does it work atm?

Comment (yuliaveres) [reply Toshi/Supporting Files/AppDelegate.swift]: it has blur effect on top, 

```
let effectView = UIVisualEffectView(effect: UIBlurEffect(style: .extraLight))
window.addSubview(effectView)
```

File: Toshi/Supporting Files/AppDelegate.swift [hunk 2]
```diff
@@ -83,21 +82,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func applicationWillResignActive(_ application: UIApplication) {
 
+        DispatchQueue.main.async {
+            self.activateScreenProtection()
```
Comment (designatednerd) [inline Toshi/Supporting Files/AppDelegate.swift]: why does this need to get dispatched on the main queue? Won't this come in on the main queue?

File: Toshi/Supporting Files/AppDelegate.swift [hunk 3]
```diff
@@ -284,22 +282,30 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         effectView.bottomAnchor.constraint(equalTo: window.bottomAnchor).isActive = true
         effectView.rightAnchor.constraint(equalTo: window.rightAnchor).isActive = true
```
Comment (designatednerd) [inline Toshi/Supporting Files/AppDelegate.swift]: ^ worth converting some of these to use `TinyConstraints` while you're mucking around in here?

File: Toshi/Supporting Files/AppDelegate.swift [hunk 4]
```diff
@@ -82,31 +102,26 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func applicationWillResignActive(_ application: UIApplication) {
+        DispatchQueue.main.async {
+            self.activateScreenProtection()
+        }
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Supporting Files/AppDelegate.swift]: so we had issues before, sometimes race conditions were happening and the activation / deactivation just did not work actually. by scheduling this to next runloop we are sure is going to work 🤷‍♂️  what can i say else in this cases 🤷‍♂️  🙈

**Pull request merged**